src/call_log/call_log.c

```diff
@@ -52,4 +52,152 @@ __wt_conn_call_log_teardown(WT_SESSION_IMPL *session)
+ */
+static int
+__call_log_print_start(WT_SESSION_IMPL *session, const char *class_name, const char *method_name)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    WT_RET(__wt_fprintf(session, conn->call_log_fst,
+      "{\n"
+      "    \"Operation\" : {\n"
+      "        \"ClassName\" : \"%s\",\n"
+      "        \"MethodName\" : \"%s\",\n",
+      class_name, method_name));
+
+    return (0);
+}
+
+/*
+ * __call_log_print_input --
+ *     Helper function for printing the input section of the call log entry. A variable number of
+ *     arguments is accepted since each API call may use a different number of inputs. Inputs are
+ *     expected to be a string in JSON format to be appended to the input list in the JSON API call
+ *     entry.
+ */
+static int
+__call_log_print_input(WT_SESSION_IMPL *session, int n, ...)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    va_list valist;
+
+    conn = S2C(session);
+
+    va_start(valist, n);
+
+    WT_ERR(__wt_fprintf(session, conn->call_log_fst, "        \"Input\" : {\n"));
+
+    for (int i = 0; i < n; i++) {
+        WT_ERR(
+          __wt_fprintf(session, conn->call_log_fst, "            %s,\n", va_arg(valist, char *)));
+    }
+
+    WT_ERR(__wt_fprintf(session, conn->call_log_fst, "        },\n"));
+
+err:
+    va_end(valist);
+
+    return (ret);
+}
+
+/*
+ * __call_log_print_output --
+ *     Helper function for printing the output section of the call log entry. A variable number of
+ *     arguments is accepted since each API call may use a different number of outputs. Outputs are
+ *     expected to be a string in JSON format to be appended to the output list in the JSON API call
+ *     entry.
+ */
+static int
+__call_log_print_output(WT_SESSION_IMPL *session, int n, ...)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    va_list valist;
+
+    conn = S2C(session);
+
+    va_start(valist, n);
+
+    WT_ERR(__wt_fprintf(session, conn->call_log_fst, "        \"Output\" : {\n"));
+
+    for (int i = 0; i < n; i++) {
+        WT_ERR(
+          __wt_fprintf(session, conn->call_log_fst, "            %s,\n", va_arg(valist, char *)));
+    }
+
+    WT_ERR(__wt_fprintf(session, conn->call_log_fst, "        },\n"));
+
+err:
+    va_end(valist);
+
+    return (ret);
+}
+
+/*
+ * __call_log_print_return --
+ *     Helper function for printing the return section of the call log entry. Each return section of
+ *     the call log entry expects to have the return value of the API call and an error message if
+ *     it exists.
+ */
+static int
+__call_log_print_return(WT_SESSION_IMPL *session, int ret_val, const char *err_msg)
+{
+    WT_CONNECTION_IMPL *conn;
+
+    conn = S2C(session);
+
+    WT_RET(__wt_fprintf(session, conn->call_log_fst,
+      "        \"Return\" : {\n"
+      "            \"ReturnVal\" : %d,\n"
+      "            \"errMsg\" : \"%s\"\n"
+      "        }\n"
+      "    }\n"
+      "}",
```

Comment:
We'll need to handle the case for appending new call log entries as well as printing the last one. Since we only have one entry for now I've left it out so that the JSON parser has no complaints.